src/coreclr/vm/stackwalk.cpp

```diff
@@ -2867,12 +2867,9 @@ void StackFrameIterator::ProcessCurrentFrame(void)
                 {
                     // We have hit the InterpreterFrame while we were not processing the interpreter frames.
                     // Switch to walking the underlying interpreted frames.
-                    PTR_InterpMethodContextFrame pTOSInterpMethodContextFrame = ((PTR_InterpreterFrame)m_crawl.pFrame)->GetTopInterpMethodContextFrame();
                     PREGDISPLAY pRD = m_crawl.GetRegisterSet();
-                    SetIP(pRD->pCurrentContext, (TADDR)pTOSInterpMethodContextFrame->ip);
-                    SetSP(pRD->pCurrentContext, dac_cast<TADDR>(pTOSInterpMethodContextFrame));
-                    SetFP(pRD->pCurrentContext, (TADDR)pTOSInterpMethodContextFrame->pStack);
-                    pRD->pCurrentContext->ContextFlags = CONTEXT_CONTROL;
+                    ((PTR_InterpreterFrame)m_crawl.pFrame)->SetContextToInterpMethodContextFrame(pRD->pCurrentContext);
```

Comment:
Is this going to work for nested stackwalks in the presence of nested exceptions? In other words, are we always going to stash the exact same m_interpExecMethodIP/SP values in the InterpreterFrame?

It feels like this wants to be a state on the stackwalker instead.

Reply:
The `SetContextToInterpMethodContextFrame` gets the topmost context of the interpreted frames in the `InterpExecMethod` belonging to the `InterpreterFrame`. These IP/SP are not stashed in the frame.

At the point where the stack walker reaches the `InterpreterFrame`, it needs to enumerate all the underlying interpreted frames and this just sets the pRD->pCurrentContext to the topmost one of them for that `InterpExecMethod`

Reply:
Would it make sense to add some asserts to guarantee that nested or parallel stackwalks are not trying to store conflicting values into `m_interpExec...` fields?